docgen/src/guides/angular-integration.md

```diff
@@ -0,0 +1,366 @@
+---
+title: Integrate with Angular2+
+mainTitle: Guides
+layout: main.pug
+name: integrate-angular
+withHeadings: true
+navHeight: 20
+editable: true
+githubSource: docgen/src/guides/angular-integration.md
+---
+
+## Introduction
+
+In this guide, we will learn together step by step how to integrate InstantSearch.js into an Angular2+ application.
+
+At the end of this guide you will have an Angular2+ application running with an InstantSearch.js search experience!
+
+For the rest of the guide we assume that you have some Angular2 knowledge and the angular-cli installed on your machine. If it's not the case have a look at the awesome lessons on [egghead.io](https://egghead.io/lessons/angular-2-say-hello-world-to-angular-2).
+
+If you are new to InstantSearch.js, best is to read first the [getting started](https://community.algolia.com/instantsearch.js/v2/getting-started.html) guide.
+
+Before going through this guide, it's well recommended to read the [customize widgets](https://community.algolia.com/instantsearch.js/v2/guides/customization.html) and the [introduce to connectors](https://community.algolia.com/instantsearch.js/v2/connectors.html) guides for InstantSearch.js
+
+## Create an Angular app
+
+We will create a default Angular application using the angular-cli, this is done like this:
+
+```sh
+$ ng new [project-name]
+$ cd [project-name]
+```
+
+Later, we can serve the application with `$ ng serve`.
+
+## Install InstantSearch.js
+
+First of all, you need to add InstantSearch.js to your dependencies:
+
+```sh
+$ yarn add instantsearch.js
+// OR
+$ npm install instantsearch.js --save
+```
+
+After you have installed InstantSearch.js to your application dependencies, we need to instantiate the InstantSearch.js library into an Angular service. So we will generate this new service:
+
+```sh
+$ ng generate service services/instantsearch
+// OR
+$ ng g s services/instantsearch
+```
+
+Open the new generated file `src/app/services/instantsearch.service.ts` and here initialize InstantSearch.js
+
+In order to initialize InstantSearch.js, you need an Algolia account with a configured and non-empty index. For the sake of simplicity, let's use the credentials of our test index:
+
+```javascript
+import { Injectable } from '@angular/core';
+import instantsearch from 'instantsearch.js/es';
+
+@Injectable()
+export class InstantSearchService {
+  search = instantsearch({
+    appId: 'latency',
+    apiKey: '3d9875e51fbd20c7754e65422f7ce5e1',
+    indexName: 'bestbuy',
```

Comment:
Hmm we need to update the attribute name and template of the hits if we do so, this not mandatory I think.